rust/src/tx_builder.rs

```diff
@@ -399,10 +456,117 @@ impl TransactionBuilder {
         };
     }
 
+    /// Set explicit auxiliary data via an AuxiliaryData object
+    /// It might contain some metadata plus native or Plutus scripts
     pub fn set_auxiliary_data(&mut self, auxiliary_data: &AuxiliaryData) {
         self.auxiliary_data = Some(auxiliary_data.clone())
     }
 
+    /// Set metadata using a GeneralTransactionMetadata object
+    /// It will be set to the existing or new auxiliary data in this builder
+    pub fn set_metadata(&mut self, metadata: &GeneralTransactionMetadata) {
+        let mut aux = self.auxiliary_data.as_ref().cloned().unwrap_or(AuxiliaryData::new());
+        aux.set_metadata(metadata);
+        self.set_auxiliary_data(&aux);
+    }
+
+    /// Add a single metadatum using TransactionMetadatumLabel and TransactionMetadatum objects
+    /// It will be securely added to existing or new metadata in this builder
+    pub fn add_metadatum(&mut self, key: &TransactionMetadatumLabel, val: &TransactionMetadatum) {
+        let mut metadata = self.auxiliary_data.as_ref()
+            .map(|aux| { aux.metadata().as_ref().cloned() })
+            .unwrap_or(None)
+            .unwrap_or(GeneralTransactionMetadata::new());
+        metadata.insert(key, val);
+        self.set_metadata(&metadata);
+    }
+
+    /// Add a single JSON metadatum using a TransactionMetadatumLabel and a String
+    /// It will be securely added to existing or new metadata in this builder
+    pub fn add_json_metadatum(
+        &mut self,
+        key: &TransactionMetadatumLabel,
+        val: String,
+    ) -> Result<(), JsError> {
+        self.add_json_metadatum_with_schema(key, val, MetadataJsonSchema::NoConversions)
```

Comment:
I'm not entirely sure if it's best to default to `NoConversions` or not. On one hand, it's stricter than the IOHK ones so people might be surprised if it fails but maybe that's better than people who aren't aware of those IOHK schemas having it do some transformation they don't want.

Reply:
I was more surprised to find out `NoConversions` and `BasicConversions` are end up being exactly the same thing.